Odata-docs/odatalib/whats-new/odatalib-8.md

```diff
@@ -0,0 +1,429 @@
+
+### Serialize `variable` scale in lowercase
+
+When writing the `Scale` attribute in XML CSDL, use `variable` in lowercase instead of `Variable`. An enum flag, `ODataLibraryCompatibility.UseLegacyVariableCasing`, was added to support the legacy behavior if desired.
+
+### Changed JSON reader and writer interfaces
+
+This is relevant if you implemented custom JSON reader or JSON writer interfaces.
+
+In OData .NET 7, there were many interfaces required to implement a custom JSON reader or write because the synchronous, asynchronous and streaming APIs were split.
+In OData .NET 8 these different APIs have been consolidated. There's now only one `IJsonWriter` interface and only one `IJsonReader` interfaces that a customer writer or reader
+would need to implement.
+
+OData .NET 7 also had two interfaces for JsonWriter factories, `IJsonWriterFactory` which accepted a `TextWriter` and `IStreamBasedJsonWriterFactory` accepted a `Stream`.
+OData .NET 8 exposes a single factory interface, `IJsonWriterFactory` which accepts a `Stream`.
+
+### Added `ODataMessageWriterSettings.ShouldIncludeAnnotation` for custom instance annotations
+
+In OData .NET 7, custom instance annotations would not be included in the written output if they were not inluded in the `@odata.include-annotations` preference token in preference header.
+However, in some cases, customers want to add custom instance annotations to the payload that should always be written regardless of end-user preference.
+
+In OData .NET 8, we introduced a `ShouldIncludeAnnotation` proeprty to the `ODataMessageWriterSettings` that allows developers to force a custom instance annotation to be written. It works as follows:
+
+- The property is a `Func<string, bool>` that receives an annotation name and returns true if the annotation should be written
+- If the writer was going to skip a certain annotation (e.g. because it was not part of the annotations filter), it will check this delegate before skipping the annotation. If the delegate returns `true`, then the annotation will not be skipped.
+
+It is important to note that returning false from the delegate does not guarantee that the annotation will not be written. For example, if an annotation was included in the preference header annotations filter, it may still be written even when the delegate returns false.
+
+```csharp
+writerSettings.ShouldIncludeAnnotation == (name) => name.StartsWith("important.");
+```
+
+### Changed `ODataBinaryStreamValue` to close underlying stream by default
+
+In OData .NET 7, when `ODataBinaryStreamValue` class is initialized using the `ODataBinaryStreamValue(Stream)` constructor, the stream is left open by default upon the object being disposed.
+
+In OData .NET 8, the stream is closed by default when the object objects is disposed. `The ODataBinaryStreamValue(Stream, bool)` constructor overload may be used when leaving the stream open is intended.
```

Comment:
I think you meant `the object is disposed` and not `the object objects is disposed` here.
```suggestion
In OData .NET 8, the stream is closed by default when the object is disposed. `The ODataBinaryStreamValue(Stream, bool)` constructor overload may be used when leaving the stream open is intended.
```

Reply:
Thanks.